### PR TITLE
Docs - Updating to pull the Hugo theme dynamically

### PR DESCRIPTION
**Pull request by artberger**
Need to bump the Hugo theme version to pick up the latest. Instead of manually updating each time, changing it to pull the latest release for that repo. For more information, see https://github.com/solo-io/doctopus/issues/28

# Checklist:

- N/A I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- N/A If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- N/A I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- N/A I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by artberger** [comment]
Confirmed formatting changes from theme update are rendering in the preview, such as these H3s here: https://gloo-edge--pr5300-adb-hugo-up-dzbdc53w.web.app/gloo-edge/latest/introduction/observability/

image

File: docs/Makefile [hunk 1]
```diff
@@ -10,7 +10,7 @@ VERSION ?= $(shell echo $(TAGGED_VERSION) | cut -c 2-)
 
 HUGO_VERSION := 0.69.2
 
-SOLO_HUGO_THEME_REVISION := 41587dc9dc719bd22cedd9803994f756e25f571f
+SOLO_HUGO_THEME_REVISION := $(curl --silent "https://api.github.com/repos/solo-io/hugo-theme-soloio/releases/latest" | jq -r .tag_name)
```
Comment (saiskee) [inline docs/Makefile:13]: Let's change this to 
```
curl --silent “https://api.github.com/repos/solo-io/hugo-theme-soloio/releases/latest” | jq -esr ‘if . == [] then null else .[] | .tag_name end’
```

else if the `curl` fails, jq will swallow the error and silently fail here.

**Comment by sam-heilbron** [comment]
What is the effect that has if breaking changes are introduced into the hugo theme?

**Comment by artberger** [comment]
> What is the effect that has if breaking changes are introduced into the hugo theme?

@sam-heilbron I am not sure , and @saiskee can correct me if I'm misunderstanding the doc build process, but a couple thoughts are:
* Before publishing a latest Solo Hugo theme release that these docs would call, we should test changing the theme to the latest Solo Hugo commit in the Edge, Mesh, & Portal docs locally to make sure the release doesn't cause any issues.
* Whatever the breaking change is would get picked up only in the master branch, which isn't the default view of the doc and is for active development, before it would be backported to a previous branch, like 1.8/latest.

But maybe there are better ways to prevent?

**Review by sam-heilbron — CHANGES_REQUESTED** [review]
There are comments on PRs in other repos (https://github.com/solo-io/gloo-mesh-enterprise/pull/1219#discussion_r704789137) that this change will not work as expected.

**Comment by artberger** [comment]
@sam-heilbron Sai's update fixed that! In the logs, it's now checking out the latest version. https://github.com/solo-io/gloo/pull/5300/checks?check_run_id=3623994445#step:7:277

**Review by sam-heilbron — APPROVED** [review]

**Pull request merged**